tools/setup/emoji/build_emoji

```diff
@@ -240,6 +240,62 @@ def setup_emoji_farm(cache_path, emoji_map, emoji_data):
 
     generate_sprite_css_files(cache_path, emoji_map, emoji_data)
 
+def setup_old_emoji_farm(cache_path, emoji_map):
+    # type: (Text, Dict[Text, Text]) -> None
+    # Code for setting up old emoji farm.
+    remapped_emojis = {
+        "0023": "0023-20e3",    # Hash
+        "0030": "0030-20e3",    # Zero
+        "0031": "0031-20e3",    # One
+        "0032": "0032-20e3",    # Two
+        "0033": "0033-20e3",    # Three
+        "0034": "0034-20e3",    # Four
+        "0035": "0035-20e3",    # Five
+        "0036": "0036-20e3",    # Six
+        "0037": "0037-20e3",    # Seven
+        "0038": "0038-20e3",    # Eight
+        "0039": "0039-20e3",    # Nine
+        "1f48f": "1f469-200d-2764-200d-1f48b-200d-1f468",    # Couple kiss
+        "1f491": "1f469-200d-2764-200d-1f468",  # Couple with heart
+    }
+
+    os.chdir(cache_path)
+    cache_emoji = os.path.join(cache_path, 'images', 'emoji')
+    cache_emoji_unicode = os.path.join(cache_path, 'images', 'emoji', 'unicode')
+    cache_google_64 = os.path.join(cache_path, 'images-google-64')
+
+    # Symlink zulip.png image file as well as check for symlink support.
+    try:
+        image_file_path = os.path.join(cache_google_64, 'zulip.png')
+        symlink_path = os.path.join(cache_emoji, 'zulip.png')
+        os.symlink(image_file_path, symlink_path)
+
+        symlink_path = os.path.join(cache_emoji_unicode, 'zulip.png')
+        os.symlink(image_file_path, symlink_path)
+    except OSError:
+        raise Exception("Error: Unable to create symlinks. Make sure you have permission to create symbolic links.")
+
+    for name, codepoint in emoji_map.items():
+        # `mapped_codepoint` refers to the file name of the image file in the new
+        # emoji farm that is same as the image in the old emoji farm.
+        mapped_codepoint = codepoint
+        if codepoint in remapped_emojis:
+            mapped_codepoint = remapped_emojis[codepoint]
+        try:
+            # `emoji_map` contains duplicate entries for the same codepoint with
+            # different names. So creation of symlink for <codepoint>.png may throw
+            # `FileExistsError`. Symlink for name file needs to be created first so
+            # that existence of symlink for corresponding <codepoint>.png doesn't
+            # hamper creation of named image file.
+            image_file_path = os.path.join(cache_google_64, '{}.png'.format(mapped_codepoint))
+            symlink_path = os.path.join(cache_emoji, '{}.png'.format(name))
+            os.symlink(image_file_path, symlink_path)
+
+            symlink_path = os.path.join(cache_emoji_unicode, '{}.png'.format(codepoint))
+            os.symlink(image_file_path, symlink_path)
```

Comment:
Presumably it's just this line that can throw a `FileExistsError`; should the try/except be here instead?

Also, call it `unicode_symlink_path` for clarity?